app/assets/javascripts/location_search.js

```diff
@@ -2,26 +2,63 @@ window.ST = window.ST || {};
 
 (function(module) {
 
-  module.initializeLocationSearch = function() {
+  module.initializeLocationSearch = function(minimumDistanceMax) {
     var searchInput = document.getElementById('q');
     var statusInput = document.getElementById('ls');
     var coordinateInput = document.getElementById('lc');
     var boundingboxInput = document.getElementById('boundingbox');
+    var maxDistanceInput = document.getElementById('distance_max');
     var homepageForm = document.getElementById('homepage-filters');
     var autocomplete = new window.google.maps.places.Autocomplete(searchInput, { bounds: { north: -90, east: -180, south: 90, west: 180 } });
     autocomplete.setTypes(['geocode']);
 
     boundingboxInput.value = null;
 
+    function toRadians(degrees) {
+      return degrees * (Math.PI/180);
+    }
+
+    function computeScale(a, b) {
+      var R = 6371; // Earth's radius in km
+
+      var lat1 = a.lat();
+      var lat2 = b.lat();
+      var lng1 = a.lng();
+      var lng2 = b.lng();
+      var lat1InRadians = toRadians(lat1);
+      var lat2InRadians = toRadians(lat2);
+      var latDiffInRadians = toRadians(lat2-lat1);
+      var lngDiffInRadians = toRadians(lng2-lng1);
+
+      // The haversine formula
+      // 'a' is the square of half the chord length between the points
+      var a = Math.sin(latDiffInRadians/2) * Math.sin(latDiffInRadians/2) +
+              Math.cos(lat1InRadians) * Math.cos(lat2InRadians) *
+              Math.sin(lngDiffInRadians/2) * Math.sin(lngDiffInRadians/2);
+      // the angular distance in radians
+      var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
+      // distance between coordinates
+      var d = R * c;
+      return d/2;
+    };
+
+    function updateViewportData(viewport) {
+      if (viewport) {
+        var boundingboxRadius = computeScale(viewport.getNorthEast(), viewport.getSouthWest());
+        maxDistanceInput.value = boundingboxRadius;
+        boundingboxInput.value = viewport.toUrlValue();
+      } else {
+        maxDistanceInput.value = minimumDistanceMax;
```

Comment:
Minimum distance of 5 km is now enforced in two places, here and here: https://github.com/sharetribe/sharetribe/pull/2079/files#diff-8b4aa234db6cb84dd4803f9b2c78ce5fR165

I think `HomepageController` is where it belongs to, this should return `null` or `undefined` if there's no viewport.